datafusion/sqllogictest/test_files/coalesce.slt

```diff
@@ -23,7 +23,7 @@ select coalesce(1, 2, 3);
 1
 
 # test with first null
-query ?T
+query IT
 select coalesce(null, 3, 2, 1), arrow_typeof(coalesce(null, 3, 2, 1));
```

Comment:
Same as `arrow_typeof` now.
```sh
> select arrow_typeof(coalesce(null, 3, 2, 1));
+---------------------------------------------------------+
| arrow_typeof(coalesce(NULL,Int64(3),Int64(2),Int64(1))) |
+---------------------------------------------------------+
| Int64                                                   |
+---------------------------------------------------------+
```
The datatype of the initial logical plan is incorrect until TypeCoercion is executed.